<a href="https://colab.research.google.com/github/afviyanabila/airbnb-data-analysis-project/blob/main/Airbnb_Data_Analysis.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# **AIRBNB ANALYSIS**

## **Project Background**

In this case study, our client gives us a data analysis project. Our client's *objective* is to rent the right properties on the Airbnb marketplace in Singapore to get the maximum profits. The right properties will affect the number of rented properties. As a data analysts we should do analysis on Singapore Airbnb data to give our client best strategy recommendations to reach their objective.


## **Datasets**

*   Airbnb Listings Properties
*   Neighborhood Mapping
*   Airbnb Listings Reviews History



## **Import Data**

All datasets available imported from Google Drive

In [ ]:
from google.colab import drive
drive.mount('/content/drive')

Mounted at /content/drive


In [ ]:
import pandas as pd

listings = pd.read_csv('/content/drive/MyDrive/Airbnb/listings.csv', sep = ',', index_col = 'Unnamed: 0')
listings.reset_index(drop=True, inplace=True)

neighborhood = pd.read_csv('/content/drive/MyDrive/Airbnb/neighborhood.csv', sep = ',', index_col = 'Unnamed: 0')
neighborhood.reset_index(drop=True, inplace=True)

reviews = pd.read_csv('/content/drive/MyDrive/Airbnb/reviews.csv', sep = ',', index_col = "Unnamed: 0")
reviews.reset_index(drop=True, inplace=True)
reviews.rename(columns={"listing_id": "id"}, inplace=True)

## **Data Preparation & Data Cleaning**

### **Airbnb Listings Properties Dataset**

In [ ]:
listings.columns

Index(['id', 'name', 'host_id', 'host_name', 'neighbourhood', 'latitude',
       'longitude', 'room_type', 'price', 'minimum_nights',
       'availability_365'],
      dtype='object')

In [ ]:
listings.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 4161 entries, 0 to 4160
Data columns (total 11 columns):
 #   Column            Non-Null Count  Dtype  
---  ------            --------------  -----  
 0   id                4161 non-null   int64  
 1   name              4161 non-null   object 
 2   host_id           4161 non-null   int64  
 3   host_name         4161 non-null   object 
 4   neighbourhood     4161 non-null   object 
 5   latitude          4161 non-null   float64
 6   longitude         4161 non-null   float64
 7   room_type         4161 non-null   object 
 8   price             4161 non-null   int64  
 9   minimum_nights    4161 non-null   int64  
 10  availability_365  4161 non-null   int64  
dtypes: float64(2), int64(5), object(4)
memory usage: 357.7+ KB


In [ ]:
print('\n listings missing values :', listings.isna().sum().sum())
listings.drop_duplicates(inplace=True)
print('\n listings duplicated values :', listings.duplicated().sum())


 listings missing values : 0

 listings duplicated values : 0


Airbnb listings properties dataset has *11 columns* consist of *4161 entries* without any missing and duplicate values
*   id: unique Airbnb listings properties id
*   name: name of listings properties
*   host_id: listings' host unique id
*   host_name: listings' host name
*   neighbourhood: neighborhood name
*   latitude: latitude location
*   longitude: longitude location 
*   room_type: room type available
*   price: listings' price
*   minimum_nights: minimum nights rent
*   availability_365: number of days available to rent





The statistic aspects of listings properties dataset being provided

In [ ]:
listings.describe().T

count          mean           std          min  \
id                4161.0  1.354784e+17  2.653249e+17  50646.00000   
host_id           4161.0  1.559857e+08  1.356249e+08  23666.00000   
latitude          4161.0  1.344527e+00  6.696490e-02      1.24826   
longitude         4161.0  1.038152e+02  6.544995e-02    103.63530   
price             4161.0  2.145052e+02  4.549214e+02      0.00000   
minimum_nights    4161.0  5.756693e+01  6.574652e+01      1.00000   
availability_365  4161.0  2.477042e+02  1.325640e+02      0.00000   

                           25%           50%           75%           max  
id                2.507292e+07  3.990020e+07  5.319607e+07  7.208120e+17  
host_id           4.187059e+07  1.367001e+08  2.388916e+08  4.804050e+08  
latitude          1.296100e+00  1.313930e+00  1.414620e+00  1.488000e+00  
longitude         1.037695e+02  1.038404e+02  1.038550e+02  1.039777e+02  
price             6.000000e+01  1.200000e+02  2.350000e+02  1.249400e+04  
minimum_nights    1.000000e+00  9.200000e+01  9.200000e+01  1.000000e+03  
availability_365  1.260000e+02  3.220000e+02  3.610000e+02  3.650000e+02

Here's the head of listings properties dataset:

In [ ]:
listings.head()

id                                 name  host_id host_name  \
0   50646      Pleasant Room along Bukit Timah   227796   Sujatha   
1   71609  Ensuite Room (Room 1 & 2) near EXPO   367042   Belinda   
2   71896      B&B  Room 1 near Airport & EXPO   367042   Belinda   
3   71903           Room 2-near Airport & EXPO   367042   Belinda   
4  275344    15 mins to Outram MRT Single Room  1439258       Kay   

  neighbourhood  latitude  longitude     room_type  price  minimum_nights  \
0   Bukit Timah   1.33432  103.78521  Private room     80              92   
1      Tampines   1.34537  103.95887  Private room    145              92   
2      Tampines   1.34754  103.95958  Private room     85              92   
3      Tampines   1.34531  103.96100  Private room     85              92   
4   Bukit Merah   1.28836  103.81144  Private room     49              60   

   availability_365  
0               365  
1               340  
2               265  
3               365  
4               296

Boxplots made to show listings' distribution by following rules where number of days available to rent is more than zero, minimum nights rent equal or less than 365 days, along with their corresponding price

In [ ]:
import plotly.express as px
fig_map = px.scatter_mapbox(listings, lat='latitude', lon='longitude', hover_name='id',zoom=10, height=500, width=1000)
fig_map.update_layout(mapbox_style='open-street-map')
fig_map.update_layout(margin={'r':0,'t':0,'l':0,'b':0}) #r=right, t=top, l=left, b=bottom
fig_map.show()

In [ ]:
import plotly.express as px
listings = listings[listings['availability_365']>0].reset_index()
fig_dist1 = px.box(data_frame=listings, x = 'availability_365', width = 1000, height=250)
display(fig_dist1)
listings = listings[listings['minimum_nights']<=365].reset_index()
fig_dist2 = px.box(data_frame=listings, x = 'minimum_nights', width = 1000, height=250)
display(fig_dist2)
#listings = listings[listings['price']>0].reset_index()
fig_dist3 = px.box(data_frame=listings, x = 'price', width = 1000, height=250)
display(fig_dist3)

After that, we plotted a map showed listings properties distribution in Singapore

In [ ]:
fig_map = px.scatter_mapbox(listings, lat='latitude', lon='longitude', hover_name='id',zoom=10, height=500, width=1000)
fig_map.update_layout(mapbox_style='open-street-map')
fig_map.update_layout(margin={'r':0,'t':0,'l':0,'b':0})
fig_map.show()

There're still lots of listings properties actually located in the outside of Singapore, so we need to clean that data first 

In [ ]:
listings = listings[listings['latitude']<1.45328]
outer_place_index = listings[ (listings['latitude'] > 1.40472) & (listings['longitude'] < 103.6638) ].index 
listings.drop(outer_place_index , inplace=True)
outer_id_index = listings[listings['id'].isin([540170321699689740, 27248990, 27532454, 29171020, 28268681, 39757287, 39732595, 708999801528964619])].index
listings.drop(outer_id_index , inplace=True)

<ipython-input-13-fcad2bb88cf5>:3: SettingWithCopyWarning:


A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy

<ipython-input-13-fcad2bb88cf5>:5: SettingWithCopyWarning:


A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy



In [ ]:
fig_map2 = px.scatter_mapbox(listings, lat='latitude', lon='longitude', hover_name='id',zoom=10, height=500, width=1000)
fig_map2.update_layout(mapbox_style='open-street-map')
fig_map2.update_layout(margin={'r':0,'t':0,'l':0,'b':0})
fig_map2.show()

Now, we got a map of listings properties distribution in the Singapore only

In [ ]:
listings = listings.drop('level_0', axis='columns')
listings = listings.drop('index', axis='columns')
listings.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 2717 entries, 0 to 3720
Data columns (total 11 columns):
 #   Column            Non-Null Count  Dtype  
---  ------            --------------  -----  
 0   id                2717 non-null   int64  
 1   name              2717 non-null   object 
 2   host_id           2717 non-null   int64  
 3   host_name         2717 non-null   object 
 4   neighbourhood     2717 non-null   object 
 5   latitude          2717 non-null   float64
 6   longitude         2717 non-null   float64
 7   room_type         2717 non-null   object 
 8   price             2717 non-null   int64  
 9   minimum_nights    2717 non-null   int64  
 10  availability_365  2717 non-null   int64  
dtypes: float64(2), int64(5), object(4)
memory usage: 254.7+ KB


As listings properties dataset being cleaned, our data now consist of 2717 entries

### **Neighborhood Mapping Dataset**

In [ ]:
neighborhood.columns

Index(['neighbourhood_group', 'neighbourhood'], dtype='object')

In [ ]:
print('\n neighborhood missing values :', neighborhood.isna().sum().sum())
neighborhood.drop_duplicates(inplace=True)
print('\n neighborhood duplicated values :', neighborhood.duplicated().sum())


 neighborhood missing values : 0

 neighborhood duplicated values : 0


In [ ]:
neighborhood.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 55 entries, 0 to 54
Data columns (total 2 columns):
 #   Column               Non-Null Count  Dtype 
---  ------               --------------  ----- 
 0   neighbourhood_group  55 non-null     object
 1   neighbourhood        55 non-null     object
dtypes: object(2)
memory usage: 1.3+ KB


Neighborhood mapping dataset has 2 columns consist of 55 entries without any missing and duplicate values
*   neighbourhood_group: Singapore regions (Central, North, East, West, North-East)
*   neighbourhood: Singapore neighbourhood

Here's the head of neighborhood dataset:

In [ ]:
neighborhood.head()

neighbourhood_group  neighbourhood
0      Central Region         Bishan
1      Central Region    Bukit Merah
2      Central Region    Bukit Timah
3      Central Region  Downtown Core
4      Central Region        Geylang

### **Airbnb Listings Reviews History**

In [ ]:
reviews.columns

Index(['id', 'date'], dtype='object')

In [ ]:
print('\n reviews missing values :', reviews.isna().sum().sum())
reviews.drop_duplicates(inplace=True)
print('\n reviews duplicated values :', reviews.duplicated().sum())


 reviews missing values : 0

 reviews duplicated values : 0


In [ ]:
reviews.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 49012 entries, 0 to 49694
Data columns (total 2 columns):
 #   Column  Non-Null Count  Dtype 
---  ------  --------------  ----- 
 0   id      49012 non-null  int64 
 1   date    49012 non-null  object
dtypes: int64(1), object(1)
memory usage: 1.1+ MB


Airbnb listings review history dataset has 2 columns consist of 49012 entries collected from 2018-01-01 to 2022-09-22 without any missing and duplicate values
*   id: unique Airbnb listings properties id
*   date: date when listings properties started being rented



Here's the head of listings review history dataset:

In [ ]:
reviews.head()

id        date
0  71609  2018-07-14
1  71609  2019-01-06
2  71609  2019-07-27
3  71609  2019-08-11
4  71609  2019-09-07

Listings review history dataset sorted based on the date

In [ ]:
reviews.sort_values('date')

id        date
6760             14530157  2018-01-01
8413             17769862  2018-01-01
6783             15310627  2018-01-01
994               2387715  2018-01-01
2495              5889741  2018-01-01
...                   ...         ...
19692            28212124  2022-09-22
37566            42079889  2022-09-22
46070            54382047  2022-09-22
46827  580226596963352685  2022-09-22
48577  640245872878209652  2022-09-22

[49012 rows x 2 columns]

## **Data Merging**

We merged specified data into new tables for further analysis 

In [ ]:
!pip install pandasql
!pip install sqlalchemy==1.4.47

Looking in indexes: https://pypi.org/simple, https://us-python.pkg.dev/colab-wheels/public/simple/
  Preparing metadata (setup.py) ... done
  Created wheel for pandasql: filename=pandasql-0.7.3-py3-none-any.whl size=26784 sha256=73c87bd376d47692e62f6e19aef37db9814e2dd47eea2db662433b54942d91cf
  Stored in directory: /root/.cache/pip/wheels/e9/bc/3a/8434bdcccf5779e72894a9b24fecbdcaf97940607eaf4bcdf9
Successfully built pandasql
Looking in indexes: https://pypi.org/simple, https://us-python.pkg.dev/colab-wheels/public/simple/
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 1.6/1.6 MB 31.1 MB/s eta 0:00:00
  Attempting uninstall: sqlalchemy
    Found existing installation: SQLAlchemy 2.0.10
    Uninstalling SQLAlchemy-2.0.10:
      Successfully uninstalled SQLAlchemy-2.0.10


In [ ]:
import pandasql as ps
from pandasql import sqldf

First table is listings_neighborhood by joined data between listings properties dataset and neighborhood mapping dataset with columns:
*   id
*   name
*   neighbourhood_group
*   neighbourhood
*   longitude
*   latitude
*   room_type
*   minimum_nights
*   availability_365


In [ ]:
listings_neighborhood = ps.sqldf("""SELECT l.id, l.name, n.neighbourhood_group, l.neighbourhood, l.longitude, l.latitude, l.room_type, l.minimum_nights, l.price, l.availability_365 from neighborhood n join listings l on n.neighbourhood=l.neighbourhood;""")
display(listings_neighborhood)

id                                             name  \
0                4926170       Room 4B in Terrace House @ Thomson Village   
1               11128602                  Luxury 3 bedroom condo. 1389sqf   
2               13243736      Charming 2Bed-2Bath- Parking Full Apartment   
3               20865016             Lush Green Neighborhood Private Room   
4               23604711           Tranquil 2Bed/2Bath Parking -Apartment   
...                  ...                                              ...   
2712  694601798894939474              Twin Room in Jurong with Facilities   
2713  696577544642312428              Twin Room in Jurong with Facilities   
2714            20791161  YOUR entire PRIVATE LUXURY PENTHOUSE condo unit   
2715  678316332235165738         50min from Singapore City, brand new 2BR   
2716            16059067                                  Single bed room   

     neighbourhood_group            neighbourhood   longitude  latitude  \
0         Central Region                   Bishan  103.832740  1.353490   
1         Central Region                   Bishan  103.830750  1.366280   
2         Central Region                   Bishan  103.828000  1.357250   
3         Central Region                   Bishan  103.830010  1.357510   
4         Central Region                   Bishan  103.828410  1.357910   
...                  ...                      ...         ...       ...   
2712         West Region              Jurong West  103.681290  1.330217   
2713         West Region              Jurong West  103.681823  1.328534   
2714         West Region                     Tuas  103.648280  1.319470   
2715         West Region                     Tuas  103.635300  1.345280   
2716         West Region  Western Water Catchment  103.686590  1.344060   

            room_type  minimum_nights  price  availability_365  
0        Private room              92     47               356  
1     Entire home/apt             180    200               365  
2     Entire home/apt              92    209               330  
3        Private room              92     50               363  
4     Entire home/apt              92    207               329  
...               ...             ...    ...               ...  
2712  Entire home/apt               6    100               350  
2713  Entire home/apt               6    100               347  
2714  Entire home/apt              92  10286                89  
2715  Entire home/apt              92     70               362  
2716     Private room              92     51               109  

[2717 rows x 10 columns]

In [ ]:
listings_neighborhood.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 2717 entries, 0 to 2716
Data columns (total 10 columns):
 #   Column               Non-Null Count  Dtype  
---  ------               --------------  -----  
 0   id                   2717 non-null   int64  
 1   name                 2717 non-null   object 
 2   neighbourhood_group  2717 non-null   object 
 3   neighbourhood        2717 non-null   object 
 4   longitude            2717 non-null   float64
 5   latitude             2717 non-null   float64
 6   room_type            2717 non-null   object 
 7   minimum_nights       2717 non-null   int64  
 8   price                2717 non-null   int64  
 9   availability_365     2717 non-null   int64  
dtypes: float64(2), int64(4), object(4)
memory usage: 212.4+ KB


Second table is listings_reviews by joined data between listings_neighborhood table and listings review history dataset with columns:
*   date
*   month: month and year from date data
*   year: from date data
*   id
*   name
*   neighbourhood_group
*   neighbourhood
*   room_type
*   price
*   minimum_nights
*   availability_365


In [ ]:
listings_reviews = ps.sqldf("""SELECT r.date, strftime('%Y-%m', r.date) month, strftime('%Y', r.date) year, ln.id, ln.name, ln.neighbourhood, ln.neighbourhood_group, ln.room_type, ln.price, ln.minimum_nights, ln.availability_365 from reviews r join listings_neighborhood ln on r.id = ln.id""")
listings_reviews

date    month  year                  id  \
0      2018-07-14  2018-07  2018               71609   
1      2019-01-06  2019-01  2019               71609   
2      2019-07-27  2019-07  2019               71609   
3      2019-08-11  2019-08  2019               71609   
4      2019-09-07  2019-09  2019               71609   
...           ...      ...   ...                 ...   
18875  2022-09-18  2022-09  2022  712014563808839949   
18876  2022-09-13  2022-09  2022  712121357550091568   
18877  2022-09-14  2022-09  2022  712907319520281078   
18878  2022-09-15  2022-09  2022  712907319520281078   
18879  2022-09-16  2022-09  2022  712907319520281078   

                                           name    neighbourhood  \
0           Ensuite Room (Room 1 & 2) near EXPO         Tampines   
1           Ensuite Room (Room 1 & 2) near EXPO         Tampines   
2           Ensuite Room (Room 1 & 2) near EXPO         Tampines   
3           Ensuite Room (Room 1 & 2) near EXPO         Tampines   
4           Ensuite Room (Room 1 & 2) near EXPO         Tampines   
...                                         ...              ...   
18875            Queen Capsule Bed (mixed dorm)  Singapore River   
18876              Single Capsule (Shared Room)  Singapore River   
18877  Charming hostel in singapore (Boat Quay)  Singapore River   
18878  Charming hostel in singapore (Boat Quay)  Singapore River   
18879  Charming hostel in singapore (Boat Quay)  Singapore River   

      neighbourhood_group     room_type  price  minimum_nights  \
0             East Region  Private room    145              92   
1             East Region  Private room    145              92   
2             East Region  Private room    145              92   
3             East Region  Private room    145              92   
4             East Region  Private room    145              92   
...                   ...           ...    ...             ...   
18875      Central Region   Shared room     60               1   
18876      Central Region   Shared room     29               1   
18877      Central Region   Shared room     31               1   
18878      Central Region   Shared room     31               1   
18879      Central Region   Shared room     31               1   

       availability_365  
0                   340  
1                   340  
2                   340  
3                   340  
4                   340  
...                 ...  
18875               346  
18876               259  
18877               342  
18878               342  
18879               342  

[18880 rows x 11 columns]

In [ ]:
listings_reviews.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 18880 entries, 0 to 18879
Data columns (total 11 columns):
 #   Column               Non-Null Count  Dtype 
---  ------               --------------  ----- 
 0   date                 18880 non-null  object
 1   month                18880 non-null  object
 2   year                 18880 non-null  object
 3   id                   18880 non-null  int64 
 4   name                 18880 non-null  object
 5   neighbourhood        18880 non-null  object
 6   neighbourhood_group  18880 non-null  object
 7   room_type            18880 non-null  object
 8   price                18880 non-null  int64 
 9   minimum_nights       18880 non-null  int64 
 10  availability_365     18880 non-null  int64 
dtypes: int64(4), object(7)
memory usage: 1.6+ MB


## **Data Analysis & Data Visualization**

### **Listings Properties Distribution in Singapore**

#### **Listings Distribution by Neighborhood Group**

In [ ]:
fig_nbgroup_pie = px.pie(listings_neighborhood, values=listings_neighborhood['neighbourhood_group'].value_counts().values, color=listings_neighborhood['neighbourhood_group'].value_counts().index, names=listings_neighborhood['neighbourhood_group'].value_counts().index, color_discrete_sequence=['#9600ff', '#995bd5', '#bf99f2', '#9cf945', '#509724'], hole=0.25)
fig_nbgroup_pie.update_traces(textposition='inside', textfont=dict(color='black',size=15), textinfo='label+percent',pull=[0.1,0.1,0.12,0.13,0.14],rotation = 120)
fig_nbgroup_pie.update_layout(title='<b>Listings Distribution by Neighborhood Group</b>', title_font=dict(size=20))
fig_nbgroup_pie.show()

Listings distribution dominated 82.9% by Central Region with 2252 properties followed by West Region 6.48% and East Region 5.34% which also represented in the map below:


In [ ]:
import plotly.express as px
fig_nbgroup_map = px.scatter_mapbox(listings_neighborhood, lat='latitude', lon='longitude', color='neighbourhood_group', hover_name='neighbourhood_group', zoom=10, height=500, width=1000, color_discrete_sequence=['#9600ff', '#995bd5', '#bf99f2', '#9cf945', '#509724'])
fig_nbgroup_map.update_layout(mapbox_style='open-street-map')
fig_nbgroup_map.update_layout(margin={'r':0,'t':0,'l':0,'b':0})
fig_nbgroup_map.show()

Based on the map, most of listings properties distributed in the Central Region which is understandable because this region is the main metropolitan in Singapore. Therefore, listings properties in the Central Region can be easily found but its competitive level will also higher than other places.

#### **Listings Distribution by Room Type**

In [ ]:
fig_roomtype_pie = px.pie(listings, values=listings['room_type'].value_counts().values, color=listings['room_type'].value_counts().index, names=listings['room_type'].value_counts().index, color_discrete_sequence=['#995bd5', '#bf99f2', '#9cf945', '#509724'], hole=0.25)
fig_roomtype_pie.update_traces(textposition='inside', textfont=dict(color='black',size=15), textinfo='label+percent', pull=[0.05,0.05,0.1,0.05],rotation = 120)
fig_roomtype_pie.update_layout(title='<b>Listings Distribution by Room Type</b>', title_font=dict(size=20))
fig_roomtype_pie.show()

Listings distribution dominated by private room type with 48.3% followed closely by entire home/apt type with 44.2%, and the rest are hotel room 4.16% and shared room 3.35%

### **The Most Rooms Listings Properties in Singapore**

#### **Top 10 Neighborhood with The Most Rooms**

In [ ]:
top_neighborhood = listings_neighborhood.groupby(['neighbourhood_group', 'neighbourhood']).agg(count=('neighbourhood','count')).sort_values(by=['count'],ascending=False).reset_index().head(10)
top_neighborhood

neighbourhood_group  neighbourhood  count
0      Central Region        Kallang    346
1      Central Region  Downtown Core    270
2      Central Region         Outram    240
3      Central Region         Novena    181
4      Central Region     Queenstown    175
5      Central Region         Rochor    174
6      Central Region        Geylang    148
7      Central Region    Bukit Merah    140
8      Central Region   River Valley    133
9         East Region          Bedok    105

In [ ]:
fig_top_neighborhood = px.bar(top_neighborhood, x = 'neighbourhood', y = 'count', color = 'neighbourhood_group', orientation = 'v', width = 1000 )
fig_top_neighborhood.update_layout(title='<b>Top 10 Neighborhood with The Most Rooms</b>', title_font=dict(size=20))
fig_top_neighborhood.update_layout(xaxis={'categoryorder':'total descending'})
display(fig_top_neighborhood)

Correlated with Central Region having the biggest number of listing properties, 9 of 10 listings properties with the most rooms also located in Central Region followed by Bedok neighborhood in East Region. The most rooms provided in Kallang neighbourhood with 346 rooms

#### **Top 10 Host ID & Host Name with The Most Rooms**

In [ ]:
top_host = listings.groupby(['host_id','host_name']).agg(count=('host_id','count')).sort_values(by=['count'],ascending=False).reset_index().head(10)
top_host

host_id                            host_name  count
0  138649185                                Fiona    217
1  156409670                                  Tia    180
2    2413412                               Kaurus     97
3  238891646                                 Neha     88
4    8948251                                 Joey     74
5   97878860                                Chess     65
6  219550151                                  Ray     49
7   23336011                             Heritage     48
8  201775246                              Hillary     46
9   32798595  Expats Housing Provider @ Raffles P     45

In [ ]:
fig_top_host = px.bar(top_host, x = 'host_name', y = 'count', orientation = 'v', width = 1000 )
fig_top_host.update_layout(title='<b>Top 10 Host ID & Host Name with The Most Rooms</b>', title_font=dict(size=20))
fig_top_host.update_layout(xaxis={'categoryorder':'total descending'})
display(fig_top_host)

Host name with the most rooms is Fiona (host ID 138649185) by 217 rooms, then Tia with 180 rooms, while the others have less than 100 rooms (host ID 156409670)

### **Price Distribution of Listings Properties in Singapore**

#### **Overall Price Distribution of Listings**

In [ ]:
price_mean = listings['price'].mean()
price_max = listings['price'].max()
price_min = listings['price'].min()
print('\n Mean Price of Listings in Singapore :', price_mean)
print('\n Minimum Price of Listings in Singapore :', price_min)
print('\n Maximum Price of Listings in Singapore :', price_max)

import plotly.express as px
fig_price = px.box(listings, x='price', width = 1000, height=250)
display(fig_price)


 Mean Price of Listings in Singapore : 254.0912771439087

 Minimum Price of Listings in Singapore : 14

 Maximum Price of Listings in Singapore : 10286


Based on boxplot plotted above, price distribution of listings properties ranged from 14 to 10286 with mean price around 254 and median price 170. Price distribution also plotted into the map below:

In [ ]:
import plotly.express as px

fig_price_map = px.scatter_mapbox(listings_neighborhood, lat='latitude', lon='longitude', color='price', hover_name='neighbourhood_group', zoom=10, width=900)
fig_price_map.update_layout(mapbox_style='open-street-map')
fig_price_map.update_layout(margin={'r':0,'t':0,'l':0,'b':0})
fig_price_map.show()

The map showed most of listings properties having prices under 2000 all over Singapore explained the price distribution is not really influenced by the location of listing properties

#### **Price Distribution of Listings Based on Neighborhood Group**

In [ ]:
price_nb = listings_neighborhood.groupby('neighbourhood_group').agg(price_mean=('price','mean')).sort_values(by=['price_mean'],ascending=False).reset_index()
price_nb

neighbourhood_group  price_mean
0        North Region  279.738462
1         West Region  258.818182
2      Central Region  258.764654
3   North-East Region  218.417722
4         East Region  183.710345

In [ ]:
fig_price_nbdist = px.box(listings_neighborhood, x='price', y='neighbourhood_group', color='neighbourhood_group')
display(fig_price_nbdist)

In [ ]:
listings_neighborhood[listings_neighborhood['price']==listings_neighborhood['price'].max()]


id                                             name  \
2714  20791161  YOUR entire PRIVATE LUXURY PENTHOUSE condo unit   

     neighbourhood_group neighbourhood  longitude  latitude        room_type  \
2714         West Region          Tuas  103.64828   1.31947  Entire home/apt   

      minimum_nights  price  availability_365  
2714              92  10286                89

In [ ]:
listings_neighborhood[listings_neighborhood['price']==listings_neighborhood['price'].min()]

id                                               name  \
2709  669949193228817088  Rental room near NTU - Jurong west st 81 SG 64...   

     neighbourhood_group neighbourhood   longitude  latitude     room_type  \
2709         West Region   Jurong West  103.695976   1.34813  Private room   

      minimum_nights  price  availability_365  
2709              92     14               365

The highest mean price based on neighborhood group in Singapore located in North Region with 279.739, followed by West and Central Region respectively 258.818 and 258.764. Both the highest price and lowest price of listings properties located in West Region respectively priced 10286 and 14

#### **Price Distribution of Listings Based on Room Type**

In [ ]:
price_rt = listings_neighborhood.groupby('room_type').agg(price_mean=('price','mean')).sort_values(by=['price_mean'],ascending=False).reset_index()
price_rt

room_type  price_mean
0  Entire home/apt  368.638634
1       Hotel room  172.557522
2     Private room  168.519817
3      Shared room   77.296703

In [ ]:
fig_price_rtdist = px.histogram(price_rt, x = 'room_type', y = 'price_mean', color = 'room_type', width = 1000)
fig_price_rtdist.update_layout(title='<b>Price Distribution Based on Room Type</b>', title_font=dict(size=20), yaxis_title = 'price_mean')
display(fig_price_rtdist)

Based on room type available, the type with the highest mean price is entire home/apt by 368.370 believed because of facilities provided by entire home/apt more complete than the other room type. Therefore, price distribution of listings properties being affected more by room type than the location

### **Reviews History of Rented Listings Properties in Singapore**

#### **Reviews History of Rented Listings Based on Date**

In [ ]:
listings_reviews_month = listings_reviews.groupby('month').agg(count=('month','count')).sort_values(by=['month']).reset_index()

fig_lreviews_month = px.line(listings_reviews_month, x='month', y='count', title='<b>Reviews History of Rented Listings Based on Month</b>', markers=True)
display(fig_lreviews_month)

Reviews history of rented listings properties from 2018, January increased to its most peak in the late 2019 until 2020, January, but immediately decreasing until 2022, March believed because of Covid-19 pandemic. After pandemic goes down, it started increasing again from 2022, April until the last data recorded in 2022, December

In [ ]:
listing_reviews_year = listings_reviews.groupby('year').agg(count=('year','count')).sort_values(by=['year']).reset_index()

fig_lreviews_year = px.histogram(listing_reviews_year, x = 'year', y = 'count', width = 1000)
fig_lreviews_year.update_layout(title='<b>Reviews History of Rented Listings Based on Year</b>', title_font=dict(size=20), yaxis_title='reviews_history')
display(fig_lreviews_year)

Correlated with reviews history based on month above, 2019 become the year with the most review history, followed by 2018 and 2022, while 2020 and 2021 having the lowest number of review history because of Covid-19 pandemic.

#### **Reviews History of Rented Listings Based on Neighborhood**

In [ ]:
listings_reviews.groupby(['neighbourhood_group','neighbourhood']).agg(neighbourhood_reviews=('neighbourhood','count')).sort_values(by=['neighbourhood_reviews'],ascending=False).reset_index()

neighbourhood_group            neighbourhood  neighbourhood_reviews
0       Central Region                  Kallang                   2563
1       Central Region                   Outram                   2272
2       Central Region            Downtown Core                   2270
3       Central Region                   Rochor                   2115
4       Central Region                  Geylang                   1695
5          East Region                    Bedok                   1067
6       Central Region             River Valley                    939
7          East Region                 Tampines                    744
8       Central Region                   Novena                    535
9       Central Region              Bukit Merah                    529
10      Central Region          Singapore River                    443
11      Central Region               Queenstown                    422
12   North-East Region                Serangoon                    395
13      Central Region                   Newton                    310
14   North-East Region                  Hougang                    297
15         West Region              Jurong West                    285
16      Central Region                  Orchard                    246
17        North Region                Woodlands                    238
18      Central Region              Bukit Timah                    237
19         East Region                Pasir Ris                    181
20      Central Region            Marine Parade                    176
21   North-East Region                  Punggol                    175
22      Central Region             Marina South                    105
23      Central Region                   Museum                    103
24         West Region              Jurong East                    101
25         West Region                 Clementi                    100
26      Central Region                   Bishan                     85
27      Central Region                  Tanglin                     73
28         West Region            Choa Chu Kang                     35
29      Central Region         Southern Islands                     27
30      Central Region                Toa Payoh                     25
31        North Region                   Yishun                     19
32         West Region              Bukit Batok                     19
33   North-East Region                 Sengkang                     17
34        North Region                Sembawang                     12
35         West Region            Bukit Panjang                     12
36   North-East Region               Ang Mo Kio                      9
37        North Region  Central Water Catchment                      4

In [ ]:
reviews_nb = listings_reviews.groupby("neighbourhood_group").agg(neighbourhood_reviews=("neighbourhood_group","count")).sort_values(by=["neighbourhood_reviews"],ascending=False).reset_index()
reviews_nb

neighbourhood_group  neighbourhood_reviews
0      Central Region                  15170
1         East Region                   1992
2   North-East Region                    893
3         West Region                    552
4        North Region                    273

In [ ]:
fig_reviews_nbdist = px.bar(reviews_nb, x='neighbourhood_reviews', y='neighbourhood_group', color='neighbourhood_group')
display(fig_reviews_nbdist)

Listings properties with the most reviews history located in Kallang, Central Region with 2563 reviews as this place also have the most rooms listings properties in Singapore. 
From 10 neighbourhood with the most reviews history, 8 of them located in the Central Region. Total of reviews history for listings properties in Central Region is 15170. This result convinced by the fact most of listings properties distributed in Central Region of Singapore. The list then followed by East, North-East, West, and North Region in sequence 

#### **Reviews History of Rented Listings Based on Listings Properties Name**

In [ ]:
top_reviews = listings_reviews.groupby(['id','name']).agg(count=('name','count')).sort_values(by=['count'],ascending=False).reset_index().head(10)
top_reviews

id                                               name  count
0   1024986                               Super Host Apartment    218
1  43337094              Double Room, DAYUSE, 5 hours: 4PM-9PM    209
2  47945468  Single cabin shared bathroom in Tanjong Pagar MRT    185
3  12162272    Great location. A spacious, well equipped home.    184
4  21415749     Luxury Capsule (Single - Mixed) [Kampong Glam]    170
5   2129215                   Luxuriously Spacious Studio Apt.    167
6  22051870                            Small Utility Cosy Room    159
7   8277151  kitchen, washer, spacious serviced apartment town    148
8  22745179                          Sea View Studio @ Shenton    143
9  15792808                         Chinatown Studio No window    137

In [ ]:
fig_top_reviews = px.bar(top_reviews, x = 'count', y = 'name', orientation = 'h', width = 1000 )
fig_top_reviews.update_layout(title="<b>Top 10 Listings with The Most Reviews</b>", title_font=dict(size=20))
fig_top_reviews.update_layout(yaxis={'categoryorder':'total ascending'})
display(fig_top_reviews)

## **Conclusion**

*   From 2018, January to 2022, September recorded that Airbnb listings properties in Singapore distributed mainly in Central Region, main metropolitan region in Singapore covered 82.9% from overall listings distribution available

*   Based on room type, the most popular listings properties are entire home/apt and private room type

*   Most of listings properties having prices under 2000 without regards of their location in Singapore. While the mean price is 254, the mean price for each neighborhood group is also not far from it

*   The highest price held by entire home/apt with mean price 368.370 followed by hotel room, private room, and shared room respectively have mean price under 200 considered the difference of facilities they may provided

*   Reviews history of listings properties was having its peak until late 2019 and immediately decreasing in number because of Covid-19 pandemic, gladly this sector started to goes up again since pandemic goes down in 2022, April

*   Correlated with Central Region has the most listings properties distribution in Singapore, the highest reviews history also come from Central Region

The recommendation we could give for our clients' from the result of analysis is to consider Airbnb properties who has price under 2000 which most of listings properties have and focused on entire home/apt and private room type which have the most demand. Furthermore, for strategic recommendations to compete with competitors, we still need further analysis especially on the correlation of price and facilities provided by listings properties that will make ours stands out and beneficial more.



